Upgrade to BU 2.0.1

alastair opened the pull request:
# Problem

New release of BU 2.0, which includes some changes to the cache API.


# Action

I ran a full spotify import cycle locally and things seem to work (counts changing as expected - this was a core bit of what used the redis integration), but we should double-check that there are no other bits of code that I didn't touch that have also changed.

pep8speaks commented:
Hello @alastair! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`listenbrainz/listenstore/redis_listenstore.py`](https://github.com/metabrainz/listenbrainz-server/blob/0654dc074b3880d9bcb236a595e2a494f23810f4/listenbrainz/listenstore/redis_listenstore.py):

> [Line 85:131](https://github.com/metabrainz/listenbrainz-server/blob/0654dc074b3880d9bcb236a595e2a494f23810f4/listenbrainz/listenstore/redis_listenstore.py#L85): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 130 characters)

alastair commented on listenbrainz/listenstore/redis_listenstore.py:
```diff
@@ -77,20 +77,20 @@ def update_recent_listens(self, unique):
         # Don't take this very seriously -- if it fails, really no big deal. Let is go.
         if recent:
-            cache._r.zadd(cache._prep_key(self.RECENT_LISTENS_KEY), recent, nx=True)
+            cache._r.zadd(cache._prep_key(self.RECENT_LISTENS_KEY, cache._glob_namespace), recent, nx=True)
```
these changes are wrong - prep_key adds the global namespace to the beginning of the key always, this argument should be None

alastair commented:
Just re-pushed to add BU 2.0.1 which allows the `namespace` argument to `_prep_key` to be None. 

I updated some calls to `cache._r.incr` to `cache.increment` because we also added the `amount` parameter. Tests now pass (including some which previously failed), but I want to double-check these cases a bit more carefully to ensure that cache values are incrementing as expected.

mayhem commented:
I fixed the merge conflict and merge master in, lets see if that fixes the tests. It would be nice to merge this so I can add more stats to other parts.
